hw_context_manager.py

```diff
@@ -0,0 +1,67 @@
+import os, time
+from contextlib import contextmanager, ContextDecorator
+
+
+#Задача-1
+
+#Создать объект менеджера контекста который будет переходить в папку которую
+# он принимает на вход.
+# Так же ваш объект должен принимать исключение которое он будет подавлять
+# Если флаг об исключении отсутствует, исключение должно быть поднято.
+
+class DirectoryChanger:
+    def __init__(self, path, exception):
+        self._current_folder = os.getcwd()
+        self._path = path
+        self._exception = exception
+
+    def __enter__(self):
+        try:
+            os.chdir(self._path)
+        except FileNotFoundError:
+            print('Folder does not exist')
+
+    def __exit__(self, type, value, traceback):
+        os.chdir(self._current_folder)
+        if type is self._exception:
+            return True
+
+os.chdir(r'C:')
+with DirectoryChanger(r'G:\Python_pro', FileNotFoundError):
+    print(os.getcwd())
+
```

Comment:
you should send an error to the context manager.
Example :

with cd('tmp', FileNotFoundError) that FileNotFoundError should be suppressed
or
with cd('tmp.txt', NotADirectoryError) that NotADirectoryError should be suppressed

thus, FileNotFoundError shouldn't be hard-coded. If you do not pass exception to context manager, it should be raised